### ISSUE TITLE
EsLint 3.5, arrow-parens Error when using async arrow function

### ISSUE THREAD

**Issue by sulliwane** [eslint/eslint#7201]
**Tell us about your environment**

* **ESLint Version: 3.5**
* **Node Version: 6.2.0**
* **npm Version: 3.10.6**

**What parser (default, Babel-ESLint, etc.) are you using?**
```javascript
{
  "parser": "babel-eslint",
  "extends": "airbnb", (v11.1.0)
  "globals": {
    "__DEV__": true
  },
  "rules": {
    "no-confusing-arrow": 0,
    "react/jsx-quotes": 0,
    "jsx-quotes": [2, "prefer-double"],
    "no-param-reassign": [2, {"props": false}],
    "no-console": 0,
    "consistent-return": 0,
    "generator-star-spacing": 0,
    "require-yield": 0,
    "no-underscore-dangle": ["error", { "allow": ["_id"] }],
    no-mixed-operators: ["error", {"allowSamePrecedence": true}],
  }
}
```
**What did you do? Please include the actual source code causing the issue.**
![screenshot from 2016-09-21 10-43-35](https://cloud.githubusercontent.com/assets/1733686/18696199/df42ffc8-7fe9-11e6-88ae-fcbcfa6c562b.png)

Is that normal ?

Many thanks!

**Comment by mysticatea** [comment]
Thank you for this issue.

It has been fixed on master. :smile: 
More info: #7101

**Comment by sulliwane** [comment]
Ok. Many thanks for your fast reply :+1:
